File: distance-service/src/main/java/pl/edu/pg/benchmarking/distance/controller/DistanceController.java
package pl.edu.pg.benchmarking.distance.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import pl.edu.pg.benchmarking.distance.entity.Point;
import pl.edu.pg.benchmarking.distance.dto.DistanceResponse;
import pl.edu.pg.benchmarking.distance.dto.DistanceRequest;
import pl.edu.pg.benchmarking.distance.service.SphericalDistanceService;

import java.util.List;

@RestController
@RequestMapping("api/distance")
public class DistanceController {

    private final SphericalDistanceService distanceService;

    public DistanceController(SphericalDistanceService distanceService) {
        this.distanceService = distanceService;
    }

    @PostMapping
    public ResponseEntity<DistanceResponse> calculateDistance(@RequestBody DistanceRequest request) {
        List<Point> points = DistanceRequest.dtoToEntityMapper().apply(request);
        Point point1 = points.get(0);
        Point point2 = points.get(1);

        Double distance = distanceService.calculateDistance(
                point1.getLatitude(), point1.getLongitude(),
                point2.getLatitude(), point2.getLongitude());

        return ResponseEntity.ok(DistanceResponse.entityToDtoMapper().apply(distance));
    }

}

File: Flask/monolith/website/distance/distance_controller.py
import website.distance.dto.distance_request as distance_request
import website.distance.dto.distance_response as distance_response
from website.distance.spherical_distance_service import SphericalDistanceService
from flask import Blueprint, request, jsonify


distance_views = Blueprint('distance', __name__)
distance_service = SphericalDistanceService()


@distance_views.route('/', methods=['POST'], strict_slashes=False)
def calculate_distance() -> dict:
    points = distance_request.dto_to_entity_mapper(request.json)
    point1 = points[0]
    point2 = points[1]

    distance: float = distance_service.calculate_distance(
        point1.latitude, point1.longitude, point2.latitude, point2.longitude)

    dto = distance_response.entity_to_dto_mapper(distance)
    return jsonify(dto.as_dict())
